### DISCUSSION TITLE
Why is my custom spectrum file not working?

### DISCUSSION THREAD

**Discussion by bdnkth** [nrc-cnrc/EGSnrc#796] [Questions] [answered]
Hi,

I'm trying to runs some simulations with the spectrum of our xray source. However, I'm running into this error:
`EGS_BaseSpectrum::createSpectrum: error on line 2 in spectrum file /mnt/c/Users/Kath/EGSnrc/HEN_HOUSE/spectra/egsnrc/tungsten70.spectrum`
`EGS_BaseSimpleSource::EGS_BaseSimpleSource:`
`    no spectrum was defined`
`createSource(parallel beam): the input is not sufficient to create a valid source`
`EGS_ObjectFactory::createObjects(): 1 errors occured while creating objects`
`EGS_ObjectFactory::createObjects(): an object with the name pencil_beam does not exist`
`EGS_Application::initSimulation(): source initialization failed`
`EGS_RandomGenerator::createRNG: no 'RNG definition' input`
`EGS_Application::initRNG(): using default RNG`

I have my spectrum file attached as a plain txt since github does not support the spectrum files. Can anyone help me trying to fix this problem?
[tungsten70.txt](https://github.com/nrc-cnrc/EGSnrc/files/7555219/tungsten70.txt)

**Comment by rtownson** [comment, marked as answer]
Remove the # on line 2, and then adjust the minimum energy of the first bin. Note that the energies on each line are the top edge of the energy bin.

Example:
```
Tungsten-70 custom spectrum
    71,0,1      
     1.0000e-03, 0.0000e+00
     2.0000e-03, 0.0000e+00
     3.0000e-03 0.0000e+00
     4.0000e-03 0.0000e+00
... etc...
```

**Reply by matheusrebe** [reply]
Hi everyone, 

Actually, I don't have a problem, but I don't understand how the second line of this file works very well, and I can't find it in the core manual. Does the second line of the spectrum file need to be: _number of lines, minimum energy, and normalization number_? When I use the normalization number 2, does it represent the minimum energy on the list? In this case, would it be 1.0000e-03?! I am almost certain that this is the case, but I am not entirely confident in my understanding.

**Reply by ftessier** [reply]
The format is detailed on **page 87** of the [BEAMnrc manual](https://nrc-cnrc.github.io/EGSnrc/doc/pirs509a-beamnrc.pdf). The last number is an integer to specify the "type" of spectrum, i.e., how the discrete values are to be understood (histogram, point spectrum, etc.)

**Comment by bdnkth** [comment]
Thanks, that has fixed my problem.
